docs/publishing.md

```diff
@@ -0,0 +1,105 @@
+# How to publish Jenkins agent deb PPA
+
+This document will talk about how to bootstrap a debian package repo, what are its components, 
+what they do and how to build, test, sign and publish a debian package. 
+https://github.com/canonical/jenkins-agent-deb will be used as an example.
+
+This document is based on https://wiki.debian.org/Packaging/Intro and goes more into details about
+the packaging process. It also tries to explain some hidden convention in the packaging process and
+put the packaging into the context of the Platform Engineering team.
+
+### Install necessary packages
+
+```
+sudo apt install build-essential devscripts debhelper
+```
+
+### Bootstrap the package repo
+
+1. debian/changelog
+
+```
+dch --create -v 1.0-0 --package jenkins-agent
+```
+
+2. debian/control - Contains information about your package as well as version of the underlying
+system used for packaging. Example of debian/control for jenkins-agent
+
+```
+Source: jenkins-agent
+Section: java
+Priority: optional
+Maintainer: <Maintainer> <reacheable email>
+Build-Depends: debhelper-compat (= 13)
+Standards-Version: 4.6.1
+
+Package: jenkins-agent
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: Minimal jenkins agent package.
+  Available as a standalone binary or a systemd service
+```
+
+3. <package>.service - If the package ships a systemd service, creating a file with this naming
+convention registers a systemd service when the package is installed
+
+```
+# File debian/jenkins-agent.service
+[Unit]
+Description=Jenkins inbound agent
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/jenkins-agent
+ExecStopPost=rm -rf /var/lib/jenkins/.ready
+
+[Install]
+WantedBy=multi-user.target
+```
+
+4. <package>.install - Specify how files will be installed in the host’s system. For jenkins agent
+we’d want a binary to be available in /usr/bin/jenkins-agent
+
+```
+jenkins-agent /usr/bin
+```
+
+5. debian/rules - Specifies how the package are built, used as a convenient wrapper for dh. For 
+jenkins-agent we’d want the systemd service to be disabled on start
```

Comment:
Some suggestions here:
- Add a verb to the step
- Put the relevant context on a new line below the step
- Is this file required? Might be good to clarify

My suggestion for the context:

> This file specifies how the packages are built and is used as a convenient wrapper for `dh`. The following example is the debian/rules file for the jenkins-agent package (note that we'd want the systemd service to be disabled on start):